Eliminate warning for zones with <6 surfaces and improve warning for <4 surfaces

RKStrand opened the pull request:
## Pull request overview

Fixes a defect (GitHub Issue #5261 in EnergyPlus that spit out an unnecessary and confusing warning when a zone had fewer than 6 surfaces.  In theory, a zone could be "complete" with only four surfaces and there is really no requirement that any zone have a specific number of surfaces.  The fix included eliminating the unnecessary warning and improving the text of a warning message that is shown when less than four surfaces in the interior radiant heat exchange algorithm to better explain what the point of the warning was.
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [x] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [x] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [x] Verify naming conventions and styles, memos and notes and defaults
  - [x] Open windows IDF Editor with modified IDD to check for errors
  - [x] If transition, add rules to spreadsheet
  - [x] If transition, add transition source
  - [x] If transition, update idfs
- [x] If new idf included, locally check the err file and other outputs
- [x] Documentation changes in place
- [x] ExpandObjects changes??
- [x] If output changes, including tabular output structure, add to output rules file for interfaces

RKStrand committed:
Merge branch 'develop' into 120446171-Mod-to-Warning-Less-than-6-Zone-Surface

RKStrand committed:
Fix of Reported Bug

Bug fix that included the removal of the error message that was
confusing, enhancement of another error message to be more descriptive,
and a new test suite input file to show that the removed error is gone
and the modified error message works.

RKStrand committed:
Additions and changes for unit test

New source file and modifications to other files to support the unit
test for this bug fix

RKStrand committed:
Merge branch 'develop' into 120446171-Mod-to-Warning-Less-than-6-Zone-Surface

mbadams5 committed:
Fix failing unit test

This corrects the unit test for FixViewFactorsTest so it uses allocate
instead of dim and allocates Zone.

nrel-bot-3 commented:
@RKStrand @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-2 commented:
@RKStrand @lgentile it has been 14 days since this pull request was last updated.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 120446171-Mod-to-Warning-Less-than-6-Zone-Surface

Conflicts:
	tst/EnergyPlus/unit/CMakeLists.txt

mjwitte committed:
Delete commented code and duplicate lines

mjwitte commented on testfiles/1ZoneUncontrolled3SurfaceZone.idf:
```diff
@@ -0,0 +1,397 @@
+!-Option OriginalOrderTop UseSpecialFormat
+!-NOTE: All comments with '!-' are ignored by the IDFEditor and are generated automatically.
+!-      Use '!' comments if they need to be retained when using the IDFEditor.
+!1ZoneUncontrolled3SurfaceZone.idf
```
@RKStrand Is the intent to add this file to the internal test suite?  If so, it needs a line in testfiles/CMakeLists.txt  Also, this file has warmup convergence errors, so it appears that something isn't working right with 3 surfaces.

RKStrand replied:
@mjwitte My intent was not to add a file to the test suite just for this error message so I think its inclusion in the commit was an error on my part.  The point was basically to have a file that I ran that showed the error message.  However, we don't need to clutter up the test suite for that since we have a unit test that does that now.  Did it get added when you merged?

mjwitte replied:
It was already in this branch, so you should delete it here, and add it to EnergyPlusDevSupport\DefectFiles\5261 instead.  But I'm concerned that we may need to post a new issue since this doesn't converge for warmup.

RKStrand replied:
@mjwitte I will go in and delete the file and add it to EnergyPlusDevSupport\DefectFiles\5261 as you requested.  I can also add a new defect to GitHub and assign it to myself if you haven't already done that.  I'll start a new Pivotal Tracker story as well and assign it to myself and show you as the requester.  I'm not sure why it is having convergence problems...

mjwitte commented:
@RKStrand I've merged in develop, cleaned out the commented code and some duplicates lines in energyplusfixture.cc  Testing with 1ZoneUncontrolled3SurfaceZone shows that the warnings are changed as expected, but as commented above, 1ZoneUncontrolled3SurfaceZone has warmup convergence problems.

RKStrand committed:
Eliminate unintentional input file

This test file was used initially to check the defect fix but it was
not intended as an addition to the test suite.  There is a unit test
now to test the functionality of this fix so this input file is not
needed in the test suite.
